internal/mocking/server/server.go

```diff
@@ -0,0 +1,164 @@
+package server
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/strslice"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+)
+
+type MockServer struct {
+	client     *client.Client
+	image      string
+	configFile string
+	apiToMock  string
+
+	LogCh chan AccessLogEntry
+	ErrCh chan error
+}
+
+type AccessLogEntry struct {
+	TimeStamp  string
+	Method     string
+	Path       string
+	StatusCode string
+
+	Error error
+}
+
+func New(ctx context.Context, client *client.Client, configFile, apiToMock string) (MockServer, error) {
+	const openApiMockImage = "muonsoft/openapi-mock:v0.3.1"
+
+	reader, err := client.ImagePull(ctx, openApiMockImage, types.ImagePullOptions{})
+	if err != nil {
+		return MockServer{}, fmt.Errorf("unable to pull mock server image: %w", err)
+	}
+
+	// Don't need the output so close it immediately
+	reader.Close()
+
+	return MockServer{
+		client:     client,
+		image:      openApiMockImage,
+		configFile: configFile,
+		apiToMock:  apiToMock,
+		LogCh:      make(chan AccessLogEntry),
+		ErrCh:      make(chan error),
+	}, nil
+}
+
+func (m MockServer) Start(ctx context.Context) (string, error) {
+	containerApiSpecPath := "mocking/fake-api.yaml"
+	containerMockingConfigFilePath := "/app/mocking/openapi-mock.yaml"
+
+	resp, err := m.client.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image:        m.image,
+			ExposedPorts: nat.PortSet{"8080": struct{}{}},
```

Comment:
Can you not chose a randomly unused port like this <https://stackoverflow.com/a/43425461/16361002>?

Also, the repetition of `8080` is ideal. I'd use a variable.